varcode/coding_effect.py

```diff
@@ -22,17 +22,54 @@
     Substitution,
     ComplexSubstitution,
     PrematureStop,
+    AlternateStartCodon,
     StartLoss,
     StopLoss,
     FrameShift,
     FrameShiftTruncation,
-    IncompleteTranscript,
     ThreePrimeUTR,
 )
 from .mutate import substitute, insert_after
 from .string_helpers import trim_shared_flanking_strings
 
-from Bio.Seq import Seq, CodonTable
+from Bio.Seq import Seq
+from Bio.Data import CodonTable
+
+START_CODONS = set(CodonTable.standard_dna_table.start_codons)
+STOP_CODONS = set(CodonTable.standard_dna_table.stop_codons)
+
+def translate(cds_seq):
+    """
+    Parameters
+    ----------
+    cds_seq : str or BioPython Seq
+        cDNA coding sequence
+
+    transcript : PyEnsembl.transcript
+        Transcript from which we got cds_seq, used for error messages.
+
+    Returns BioPython Seq of amino acids
+    """
+    cds_seq = Seq(cds_seq)
+
+    # In case sequence isn't a multiple of 3, then truncate it
+    truncated_cds_len = int(len(cds_seq) / 3) * 3
+    truncated_cds_seq = cds_seq[:truncated_cds_len]
+
+    # turn cDNA sequence into a BioPython sequence, translate
+    # to amino acids.
+    # passing cds=False since we may want to deal with premature
+    # stop codons
+    protein = truncated_cds_seq.translate(to_stop=True, cds=False)
+
+    if protein[0] != "M" and cds_seq[:3] in START_CODONS:
+        # TODO: figure out when these should be made into methionines
+        # and when left as whatever amino acid they normally code for
+        # e.g. Leucine start codons
+        # See: DOI: 10.1371/journal.pbio.0020397
+        return "M" + protein[1:]
```

Comment:
From a quick reading of things like 10.1126/science.1224439 and 10.1126/science.1220270 , it seems like CUG start codons will truly translate as L, whereas other cryptic start codons may initiate translation by mis-pairing with methionine tRNA's, so should be translated as M. Does that fit with what you've seen?

If that's true, it seems like this is the wrong behavior for CUG start codons (i.e. they should be left as L). Seems reasonable to also leave this as a TODO, though.